installer/I05_emba_docker_image_dl.sh

```diff
@@ -47,10 +53,32 @@ I05_emba_docker_image_dl() {
         fi
         if command -v docker > /dev/null ; then
           export DOCKER_CLI_EXPERIMENTAL=enabled
-          echo -e "${ORANGE}""EMBA docker image will be downloaded.""${NC}"
+          echo -e "${ORANGE}""Checking for EMBA docker image...""${NC}"
           echo -e "${ORANGE}""CONTAINER VARIABLE SET TO ""${CONTAINER}""${NC}"
-          docker pull "${CONTAINER}"
-          # we do not need to download the latest docker image -> we can just tag it locally
+          
+          # First, check whether the local mirror exists
+          if docker images --format "{{.Repository}}:{{.Tag}}" | grep -q "${CONTAINER}"; then
+            echo -e "${GREEN}""Found local image ${CONTAINER}, skipping download.""${NC}"
+          else
+            echo -e "${ORANGE}""Local image not found, attempting to download.""${NC}"
+            if ! docker pull "${CONTAINER}"; then
+              echo -e "${RED}""Failed to download ${CONTAINER}.""${NC}"
+              echo -e "${ORANGE}""Checking if we have any usable local images...""${NC}"
+              
+              # Check if there are any embeddedanalyzer/emba images
+              if ! docker images | grep -q "embeddedanalyzer/emba"; then
+                echo -e "${RED}""No local EMBA images found. Installation may be incomplete.""${NC}"
+                read -p "Press any key to continue anyway..." -n1 -s -r
+              else
+                echo -e "${GREEN}""Found alternative local EMBA image, will use that instead.""${NC}"
```

Comment:
As we define the needed docker image tag in the docker-compose [file](https://github.com/e-m-b-a/emba/blob/d9df12122f59db4cc47a5c4596b8b48aa34480e3/docker-compose.yml#L4C8-L4C11) the result will be a not working EMBA installation if there is no match. At least we should include another check if the already installed version matches the version from the compose file and if not we need another warning like in L71 but with a clean exit condition for the installer.